remote.rb

```diff
@@ -8,19 +8,90 @@
 
 page = Nokogiri::HTML(browser.html)
 
-data = []
+data_arr = []
+
+itter = 1
+index = 0
+post_d = nil
 
 page.css('.owwb-cs-slide-list-account-item').each do |i|
-    account  = i.css('.contract-number').text
-    balance  = i.css('.owwb-cs-slide-list-amount-value').text
-    currency = i.css('.owwb-cs-slide-list-amount-currency').text
-
-data.push(  
-    name:     account,
-    balance:  balance,
-    currency: currency
-)  
-end  
-
-puts JSON.pretty_generate(data)
+  account  = i.css('.contract-number').text
+  balance  = i.css('.owwb-cs-slide-list-amount-value').text.to_f
+  currency = i.css('.owwb-cs-slide-list-amount-currency').text
+
+  pag = []
+  i.css('li').each do |li|  
+    li.children.children.children.each{ |i| pag << i.text.strip.gsub(/\s{2,100}/, '') } 
+  end 
+
+  pag = pag.select{ |i| i =~ /[a-z, 1-8]/ }
+  nature = pag[pag.index("Produsul")+1]
+  adress = 'file:///home/zik/ruby/trans_list_' + itter.to_s + '.html'
+  browser.goto(adress)
+  itter += 1
+  transactions_page = Nokogiri::HTML(browser.html)
+
+  tr_arr_1 = []
+  tr_arr_2 = []
+  tr_arr_3 = []
+
+  transactions_page.css('tbody').each do |tbody|
+    tbody.children.each do |tr|  
+      if tr.text.strip.match(/^\d{2}\/\d{2}\/\d{2}\s*[a-zA-Z]\S*/)
+        tr_arr_1 << tr
+      elsif tr.text.strip.match(/^\d{2}\/\d{2}\/\d{2}\s{1,5}\d{2}\/\d{2}\/\d{2}/)
+        tr_arr_2 << tr
+      elsif tr.text.strip.match(/^\d{2}\/\d{2}\/\d{2}$/)
+        tr_arr_3 << tr.children.children.children.text.strip.gsub(/\"/,'')
```

Comment:
i'm sure there is a way to build css selector w/o children.children.children.children.children.children